examples/microprofile/cors/src/test/java/io/helidon/microprofile/examples/cors/TestCORS.java

```diff
@@ -0,0 +1,265 @@
+ * limitations under the License.
+ *
+ */
+package io.helidon.microprofile.examples.cors;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+
+import io.helidon.common.http.Headers;
+import io.helidon.common.http.MediaType;
+import io.helidon.config.Config;
+import io.helidon.media.common.MediaSupport;
+import io.helidon.media.jsonp.common.JsonProcessing;
+import io.helidon.microprofile.server.Server;
+import io.helidon.webclient.WebClient;
+import io.helidon.webclient.WebClientRequestBuilder;
+import io.helidon.webclient.WebClientResponse;
+import io.helidon.webserver.cors.CrossOriginConfig;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class TestCORS {
+
+    private static final String JSON_MESSAGE_RESPONSE_LABEL = "message";
+    private static final String JSON_NEW_GREETING_LABEL = "greeting";
+
+    private static final JsonBuilderFactory JSON_BF = Json.createBuilderFactory(Collections.emptyMap());
+    private static final JsonProcessing JSON_PROCESSING = JsonProcessing.create();
+
+
+    private static WebClient client;
+    private static Server server;
+
+    @BeforeAll
+    static void init() {
+        Config serverConfig = Config.create().get("server");
+        Server.Builder serverBuilder = Server.builder();
+        serverConfig.ifExists(serverBuilder::config);
+        server = serverBuilder
+                .port(-1) // override the port for testing
+                .build()
+            .start();
+        client = WebClient.builder()
+                    .baseUri("http://localhost:" + server.port())
+                    .mediaSupport(MediaSupport.builder()
+                        .registerDefaults()
+                        .registerReader(JSON_PROCESSING.newReader())
+                        .registerWriter(JSON_PROCESSING.newWriter())
+                        .build())
+                    .build();
+    }
+
+    @AfterAll
+    static void cleanup() {
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    @Order(1) // Make sure this runs before the greeting message changes so responses are deterministic.
+    @Test
+    public void testHelloWorld() throws Exception {
+
+        WebClientResponse r = getResponse("/greet");
+
+        assertEquals(200, r.status().code(), "HTTP response1");
```

Comment:
Please use Hamcrest assertions.

All Helidon tests should be written using Hamcrest (there are a few exceptions such as `fail` and `assertAll`) - see dev guidelines.

This should be
`assertThat("HTTP response1", r.status().code(), is(200));`
This is an overall problem in all the tests.

Reply:
Done. (We should bring all examples into line with this so there are no misleading patterns around people might follow.)